OpenTracing/OpenTracing.OpenTracing/ISpan.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+
+namespace OpenTracing
+{
+    public interface ISpan<T> 
+    {
+        T GetSpanContext();
+
+        void Finish();
+        void FinishWithOptions(DateTime finishTime);
+
+        void SetBaggageItem(string restrictedKey, string value);
```

Comment:
Thanks...

Also, a general point of order: I was inspired by your `SpanContext` stuff and am attempting to find a compromise where other languages can incorporate a context concept at the OpenTracing API layer without introducing tons of new terminology and cognitive load for the user. I will keep you informed of the RFC PR on that front which I hope to send out this weekend. (I've done the interesting part but need to clean things up)